src/common/libjob/specutil.c

```diff
@@ -337,62 +337,77 @@ int specutil_attr_check (json_t *o, char *errbuf, int errbufsz)
     return -1;
 }
 
-static json_t *specutil_tasks_create (json_t *argv)
+/* Create the 'tasks' section of a jobspec. */
+json_t *specutil_tasks_create (int argc, char **argv)
 {
     json_t *tasks;
+    json_t *argv_json;
 
-    if (!(tasks = json_pack ("[{s:O s:s s:{s:i}}]",
-                             "command", argv,
-                             "slot", "task",
+    if (!(argv_json = specutil_argv_create (argc, argv))) {
+        return NULL;
+    }
+    if (!(tasks = json_pack ("[{s:o s:s s:{s:i}}]",
+                             "command",
+                             argv_json,
+                             "slot",
+                             "task",
                              "count",
-                               "per_slot", 1))) {
+                             "per_slot",
+                             1))) {
         errno = ENOMEM;
+        ERRNO_SAFE_WRAP (json_decref, argv_json);
         return NULL;
```

Comment:
Minor: avoid the ERRNO_SAFE_WRAP macro by reversing the order of errno assignment and `json_decref()`.